components/graph/git-node.js

```diff
@@ -62,13 +89,21 @@ var GitNodeViewModel = function(graph, sha1) {
       }, 200);
     }
   });
-  this.branchingFormVisible = ko.observable(false);
   this.canCreateRef = ko.computed(function() {
     return self.newBranchName() && self.newBranchName().trim() && self.newBranchName().indexOf(' ') == -1;
   });
   this.branchOrder = ko.observable();
   this.aboveNode = undefined;
   this.belowNode = undefined;
+  this.refSearchFormVisible = ko.observable(false);
+  this.refSearchFormVisible.subscribe((newValue) => {
+    if (!newValue) {
+      // Small timeout because in ff the form is hidden before the submit click event is registered otherwise
+      setTimeout(() => {
+        this.refSearchFormVisible(false);
```

Comment:
I tested it in Firefox too. But for me it worked also without this code.
Also I don't quite understand why setting `refSearchFormVisible` to `false` after it is already set to `false` should do anything. Doesn't knockout check if the value is the same as before and does nothing?